gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -168,42 +139,61 @@ public static Properties loadJobConfig(Properties properties, File jobConfigFile
     }
 
     // Then load the job configuration properties defined in the job configuration file
-    jobProps.putAll(ConfigurationConverter.getProperties(new PropertiesConfiguration(jobConfigFile)));
-    jobProps.setProperty(ConfigurationKeys.JOB_CONFIG_FILE_PATH_KEY, jobConfigFile.getAbsolutePath());
+    jobProps.putAll(ConfigurationConverter.getProperties(
+        new PropertiesConfiguration(new Path("file://", jobConfigPath).toUri().toURL())));
+
+    jobProps.setProperty(ConfigurationKeys.JOB_CONFIG_FILE_PATH_KEY, jobConfigPath.toString());
     return jobProps;
   }
 
   /**
-   * Add {@link org.apache.commons.io.monitor.FileAlterationMonitor}s for the given
+   * Add {@link gobblin.util.filesystem.PathAlterationMonitor}s for the given
    * root directory and any nested subdirectories under the root directory to the given
-   * {@link org.apache.commons.io.monitor.FileAlterationMonitor}.
+   * {@link gobblin.util.filesystem.PathAlterationMonitor}.
    *
-   * @param monitor a {@link org.apache.commons.io.monitor.FileAlterationMonitor}
-   * @param listener a {@link org.apache.commons.io.monitor.FileAlterationListener}
-   * @param rootDir root directory
+   * @param monitor a {@link gobblin.util.filesystem.PathAlterationMonitor}
+   * @param listener a {@link gobblin.util.filesystem.PathAlterationListener}
+   * @param rootDirPath root directory
    */
-  public static void addFileAlterationObserver(FileAlterationMonitor monitor, FileAlterationListener listener,
-      File rootDir) {
-    // Add a observer for the current root directory
-    FileAlterationObserver observer = new FileAlterationObserver(rootDir);
+  public static void addPathAlterationObserver(PathAlterationMonitor monitor, PathAlterationListener listener,
+      Path rootDirPath)
+      throws IOException {
+    PathAlterationObserver observer = new PathAlterationObserver(rootDirPath);
     observer.addListener(listener);
     monitor.addObserver(observer);
 
-    // List subdirectories under the current root directory
-    File[] subDirs = rootDir.listFiles(new FileFilter() {
-      @Override
-      public boolean accept(File file) {
-        return file.isDirectory();
-      }
-    });
+    Configuration conf = new Configuration();
+    try (final FileSystem fileSystem = rootDirPath.getFileSystem(conf)) {
+
+      // List subdirectories under the current root directory, obtain a Path array
+      FileStatus[] subDirsStatus = fileSystem.listStatus(rootDirPath
```

Comment:
Instead of doing this, do
```java
FileStatus[] subDirsStatus = fileSystem.listStatus(rootDirPath)
Iterables.filter(Lists.newArrayList(subDirsStatus), new Predicate<FileStatus> {
  public boolean apply(FileStatus status) {
     return status.isDir();
  }
});
```
This reduces the number of rpc calls significantly.